src/indexing/reducer.js

```diff
@@ -1,6 +1,22 @@
 import imm from 'object-path-immutable';
 import { createAction, resourcesArray } from '../api/utils';
 
+/**
+ * Helpers
+ */
+class Index extends Array {
```

Comment:
So... subclassing primitives is something you can do in node - but not Babel. So this only half-works. The constructor works, but any subsequent manipulations are on Array, not Index. 

I'm debating on whether this should just live as logic in `indexArray` to make it less confusing.